build.yaml

```diff
@@ -1113,7 +1116,38 @@ job_type:
       coverage_report: true
       clean_repo: true
       timeout: 30
+      directories_to_delete: *run_trial_directories_to_delete
 
+    run_trial_on_AWS_CentOS_7_flocker.node.functional.test_docker:
+      # FLOC-3903: docker on centos use loop-devmapper
+      # by default. That makes it much slower than Ubuntu
+      # with aufs. It leads to timeouts, but seems to do
+      # a bit better on the medium instance
+      on_nodes_with_labels: 'aws-centos-7-SELinux-T2Medium'
+      module: flocker.node.functional.test_docker
+      with_steps:
+        - { type: 'shell', cli: *run_trial_cli }
+      archive_artifacts: *flocker_artifacts
+      publish_test_results: true
+      coverage_report: true
+      clean_repo: true
+      # Increase the timeout due to FLOC-3903
+      timeout: 45
+      directories_to_delete: *run_trial_directories_to_delete
```

Comment:
This stanza is the point of the branch. The rest is supporting this
change. This is largely copied from the existing run_trial stanza.